### ISSUE TITLE
Implement slash commands to enter and to list 

### ISSUE THREAD

**Issue by jesusgn90** [islos-efe-eme/auto-news#31]
We need a way to mark yourself as online so other users can check if you are online or not. This also allows us to keep somehow a history / some metrics about our sessions.

**Proposal**

- [ ] Add command `/enter` to Slack bot
- [ ] Add command `/list` to Slack bot
- [ ] Add command `/enter` to Telegram bot (**blocked atm**)
- [ ] Add command `/list` to Telegram bot (**blocked atm**)
- [ ] Add lightweight shared storage, JSON based and SQL-like syntax
- [ ] Unit tests
- [ ] Integration tests

**About `/list` command**

- [ ] Should return the list of users that executed `/enter` in the last 4 hours.
- [ ] The result should be the same in both Slack and Telegram as well.

**About `/enter` command**

- [ ] Should add a record including: username, date and time, source (Slack only atm)
- [ ] This command should fire a message in both Slack and Telegram letting know other users who just entered and mentioning users that didn't enter yet, asking them for entering.

**About the _shared storage_**

- [ ] It should be a lightweight solution, ideally based on a local JSON file handled by a SQL-like library (there are plenty of them out there)
- [ ] Should live at repository root level, and should be shareable by Slack, Telegram and any other future module.

Besides that, the team **reminder frequency** should be lower now:

- [ ] Ask only at 11:00, 17:00, 21:00 (UTC)

**Comment by crolopez** [comment]
It would be nice if the purpose of  `/enter` command was automated. That information is accessible through Social Club. We can see it when we start the game

I've been looking for official API for this service but no luck. Then, I have found [this guide](https://rockstar-api.readthedocs.io/en/latest/public-general.html#friends-list) which explains how to make requests to Social Club and how to process some information once you are logged in. Some commands may be obsolete due to this documentation is not updated since 2018.

However, the authentification process currently needs a captcha resolution, which makes any attempt to automate a connection to the Social Club **impossible**.

Another possibility is to extract this information through a client that parses the `launcher.log` file, which is normally located in `C:\Users\<your_user>\Documents\Rockstar Games\Launcher`, looking for the following line:

``` ossec.log
[2020-02-09 22:14:13.153] [DISPLAY] [15036] [gamelaunch] Launching game...
```
or

``` ossec.log
[2020-02-09 22:14:04.315] [DISPLAY] [Main ] [launcher] User 'Manolo' is now signed in
```

The client will then send the equivalent notification to `/enter`.

I think that we need a long discussion about this since this local client could provide value through the collection of other log files (if there are more). In addition, this client should be combined with the Snapmatic bot IMO.

In short, I agree with the issue.

**Comment by jesusgn90** [comment]
Re the API you mentioned, I've already explored that last weekend, and try some curls with no success, it was mainly based on scrapping and for the old Social Club, so in fact, no longer useful

**Comment by jesusgn90** [comment]
Re the local client, slightly connected to https://github.com/islos-efe-eme/auto-news/issues/33

**Comment by jesusgn90** [comment]
> The client will then send the equivalent notification to /enter.

It won't. The client can just send the POST with the payload to the running bot throught it's public domain (POST https://slackbot.appgranada.com/enter)

No need to merge too much things, no need to duplicate logic.
